### PR TITLE
Feature/stan 43

### PR DESCRIPTION
**Pull request by om-sharma-fr**
forgot password and reset password api

### CODE REVIEW

File: src/logic/v1/user/password/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * Function to update user's password
+ * @param {Object} params - request's body payload (password, user)
+ * @returns {Object|Error} - return success message or error if any
+ */
+const resetPassword = async (params = {}) => {
+  try {
+    const { password, user: userInfo } = params;
+
+    //fetch user by id
+    const user = await queries.common.user.fetchOne({ id: userInfo.id });
```
Comment (fr33range) [inline src/logic/v1/user/password/index.js:18]: Not needed as the user is being fetched when the token is being verified and passed on

File: src/logic/v1/user/password/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import communication from '../../../../shared/utils/communication';
+
+/**
+ * Function to update user's password
+ * @param {Object} params - request's body payload (password, user)
+ * @returns {Object|Error} - return success message or error if any
+ */
+const resetPassword = async (params = {}) => {
+  try {
+    const { password, user: userInfo } = params;
```
Comment (fr33range) [inline src/logic/v1/user/password/index.js:15]: use the key user directly

File: src/logic/v1/user/password/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    throw utils.common.createError(error, 'resetPasswordFailed');
+  }
+};
+
+/**
+ * Function to sent forgot password mail to register user's email ID
+ * @param {Object} params - request's body payload (username) 
+ * @returns {Object|Error} - return success message or error if any
+ */
+const forgotPassword = async (params = {}) => {
```
Comment (fr33range) [inline src/logic/v1/user/password/index.js:45]: rename function as resetRequest

File: src/route/v1/user/password/base.validation.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    .required()
+    .error(error => createValidationError(error, {
+      'any.required': validationErrors.usernameMissing,
+      'string.min': validationErrors.usernameShort,
+      'string.max': validationErrors.usernameLong
+    }))
+});
+
+export default {
+  resetPassword,
```
Comment (fr33range) [inline src/route/v1/user/password/base.validation.js:45]: rename function as "set"

File: src/logic/v1/user/password/index.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import queries from '../../../../data_interaction/internal/queries';
+import constants from '../../../../shared/constants';
+import communication from '../../../../shared/utils/communication';
+
+/**
+ * Function to update user's password
+ * @param {Object} params - request's body payload (password, user)
+ * @returns {Object|Error} - return success message or error if any
+ */
+const resetPassword = async (params = {}) => {
```
Comment (fr33range) [inline src/logic/v1/user/password/index.js:13]: rename function as "set"

File: src/logic/v1/user/password/index.js (hunk 3 above)
Comment (fr33range) [inline src/logic/v1/user/password/index.js:45]: rename function as "forgot"

File: src/logic/v1/user/password/index.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      message: constants.apiResponse.success.forgotPasswordMailSendSuccessful.message
+    };
+  } catch (error) {
+    throw utils.common.createError(error, 'forgotPasswordFailed');
+  }
+};
+
+export default {
+  resetPassword,
+  forgotPassword
```
Comment (fr33range) [inline src/logic/v1/user/password/index.js:97]: rename function as "forgot"

File: src/route/v1/user/password/base.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import urls from '../../../../shared/urls';
+import controller from '../../../../controller';
+import constants from '../../../../shared/constants';
+import validators from './base.validation'
+
+const resetPasswordRoute = {
```
Comment (fr33range) [inline src/route/v1/user/password/base.js:6]: rename variable as "setRoute"

File: src/route/v1/user/password/base.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  method: urls.v1.user.password.reset.method,
+  url: urls.v1.user.password.reset.url,
+  isAuthenticated: true,
+  tokenType: [constants.common.jwtTokenTypes.resetPassword],
+  validation: validators.resetPassword,
+  handler: (params) => controller.v1.user.password.resetPassword(params)
+};
+
+const forgotPasswordRoute = {
```
Comment (fr33range) [inline src/route/v1/user/password/base.js:15]: rename variable as "forgotRoute"

File: src/shared/constants/apiResponse.js [hunk 9]
```diff
@@ -246,6 +254,14 @@ const failure = {
     message: `Unable to invite user.`,
     code: `F-L-00058`
   },
+  forgotPasswordFailed: {
+    message: `Unable to request forgot password.`,
```
Comment (fr33range) [inline src/shared/constants/apiResponse.js:258]: unable to reset the password

File: src/shared/urls/v1/user/password/base.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export default {
+  forgot: {
+    url: `/forgot`,
+    method: `post`
+  },
+  reset: {
+    url: `/reset`,
+    method: `post`
```
Comment (fr33range) [inline src/shared/urls/v1/user/password/base.js:8]: method should be patch

Comment (om-sharma-fr) [reply src/shared/urls/v1/user/password/base.js:8]: updated the method type

File: src/route/v1/user/password/base.js (hunk 8 above)
Comment (om-sharma-fr) [reply src/route/v1/user/password/base.js:15]: updated the variable name

File: src/route/v1/user/password/base.js (hunk 7 above)
Comment (om-sharma-fr) [reply src/route/v1/user/password/base.js:6]: updated the variable name to **setRoute**

File: src/logic/v1/user/password/index.js (hunk 6 above)
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:97]: updated the function name

File: src/logic/v1/user/password/index.js (hunk 3 above)
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:45]: updated function name

File: src/logic/v1/user/password/index.js (hunk 5 above)
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:13]: updated function name to set

File: src/route/v1/user/password/base.validation.js (hunk 4 above)
Comment (om-sharma-fr) [reply src/route/v1/user/password/base.validation.js:45]: updated function name to set

File: src/logic/v1/user/password/index.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+ */
+const forgotPassword = async (params = {}) => {
+  try {
+    const { username } = params;
+
+    //check if user exist by id
+    const user = await queries.common.user.fetchOne({ username });
+
+    // If user doesn't exist, throw error
+    if (!user?.id) {
```
[the remark this answers is not in the document]
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:53]: No its forgot password api , we only need username here , its public api

File: src/logic/v1/user/password/index.js [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+ * Function to sent forgot password mail to register user's email ID
+ * @param {Object} params - request's body payload (username) 
+ * @returns {Object|Error} - return success message or error if any
+ */
+const forgotPassword = async (params = {}) => {
+  try {
+    const { username } = params;
+
+    //check if user exist by id
+    const user = await queries.common.user.fetchOne({ username });
```
[the remark this answers is not in the document]
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:50]: For forgot password, we dont have any token, its public api

File: src/logic/v1/user/password/index.js (hunk 1 above)
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:18]: updated the code

File: src/logic/v1/user/password/index.js (hunk 2 above)
Comment (om-sharma-fr) [reply src/logic/v1/user/password/index.js:15]: updated the code

**Review by fr33range — APPROVED** [review]

**Pull request merged**